### ISSUE TITLE
Create Leaderboard for NLU/Core models.

### ISSUE THREAD

**Issue by kedz** [RasaHQ/rasa#8679]
**Description of Problem**: Currently we are treating our regression tests like a leaderboard and looking at past regression test performance as a gatekeeper for a new feature. This is not ideal for a couple reasons. Regression tests are intended to determine that we haven't broken anything with a new commit, and so:

*   regression tests should be fast (we don't need to train to convergence) since they get run frequently during PRs
*   regression tests should test many combinations of components on many datasets to find bugs but are not necessarily running the optimal component combination/configuration per dataset

A good leaderboard on the other hand should have clearly defined train, validation, and test splits.

Proposed model classes (in our case this a feature pipeline+nlu model/entity extraction model or feature-pipeline+core policy pipeline) submitted to the leaderboard should train on the training split and optimize hyperparameters on the validation set on a per dataset basis.

If we are being extra thorough, test measures could be a averaged over multiple runs of the same model using different random seeds, and standard deviation of performance also reported.

Model size/training time could also be included in the leaderboard metrics.

**Overview of the Solution**: Write a proposal and get feedback on leaderboard design and then implement leaderboard design.

Some questions the proposal might offer answers to:

*   What datasets do we use?
*   What model classes do we use?
*   How often does it run?
*   Where do we host results?
*   How do we add a new model class?

**Possible Blockers**: We are still trying to figure out how to meaningfully evaluate core policies.

**Definition of Done**:

*   [ ] Write and get feedback on leaderboard proposal
*   [ ] Implement the leaderboard

**Comment by JEM-Mosig** [comment]
I'm not sure I understand your proposal. 

> should have clearly defined train, validation, and test splits

Don't we use clear splits with [this](https://github.com/RasaHQ/training-data)?

> optimize hyperparameters on the validation set on a per dataset basis
> test measures could be a averaged over multiple runs

This would make testing slower, not faster, right?

Generally, we need to know if things break with the standard configs that we defined, not whether-or-not you can find some config for which the performance on a particular dataset is better.

**Comment by amn41** [comment]
another benefit of having a public leaderboard: 
A few times a year someone will publish a paper where they benchmark Rasa on an NLU task. Usually they take a sparse-features-only config, and then show that  BERT+autoML or whatever outperforms that model (congrats!). Not a big deal but would be nice to have a table to point to with different Rasa configs and how they perform on standard benchmarks.

**Comment by koernerfelicia** [comment]
> Generally, we need to know if things break with the standard configs that we defined

This feels like a separate issue, though. I think the leaderboard doesn't/shouldn't exclude that kind of testing.

Somewhat tangential: wonder if we could have community members join in and try to rise to the top of the leaderboard. Like a kaggle competition. Just spitballing, dunno if there'd be interest in this, or what the logistics would be.
